### PR TITLE
feat(manager/gitlab-include): support multi-document package files

### LINKED ISSUE #24567: Support GitLab CI templates
### Describe the proposed change(s).

Details in https://github.com/renovatebot/renovate/discussions/24562

Reproduction repo forked to https://github.com/renovate-reproductions/24562

### PR DESCRIPTION
**Pull request by fgreinacher**
## Changes

Extract `gitlabci-include` dependencies from multi-document YAML files.

## Context

Closes https://github.com/renovatebot/renovate/issues/24567

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

---

:hammer_and_pick: with :heart: by [Siemens](https://opensource.siemens.com/)

### CODE REVIEW

File: lib/modules/manager/gitlabci-include/extract.spec.ts [hunk 1]
```diff
@@ -68,5 +69,37 @@ describe('modules/manager/gitlabci-include/extract', () => {
...
+        include:
+          - project: mikebryant/include-source-example
+            ref: 1.0.0
+        ---
+        include:
+          - project: mikebryant/include-source-example2
+            ref: 2.0.0
+        ---
+        more:
+          content: to be ignored
```
Comment (fgreinacher) [inline lib/modules/manager/gitlabci-include/extract.spec.ts:87]: Adding some documents here that won't comply with the schema. This should reduce the change of from breaking things once we add schema validation :)

File: lib/modules/manager/gitlabci-include/extract.spec.ts [hunk 2]
```diff
@@ -68,5 +69,37 @@ describe('modules/manager/gitlabci-include/extract', () => {
...
+      `;
+      const res = extractPackageFile(multiDocFile);
+      expect(res?.deps).toHaveLength(2);
```
Comment (viceice) [inline lib/modules/manager/gitlabci-include/extract.spec.ts:90]:
```suggestion
```
not needed, test below asserts array length

File: lib/modules/manager/gitlabci-include/extract.spec.ts [hunk 3]
```diff
@@ -68,5 +69,37 @@ describe('modules/manager/gitlabci-include/extract', () => {
...
+      expect(res?.deps).toMatchObject([
```
Comment (viceice) [inline lib/modules/manager/gitlabci-include/extract.spec.ts:91]:
```suggestion
      expect(res).toMatchObject([
```
prefer to match full results

File: lib/modules/manager/gitlabci-include/extract.ts [hunk 4]
```diff
@@ -72,14 +72,20 @@ export function extractPackageFile(
...
+    const docs = parseYaml<GitlabPipeline>(replaceReferenceTags(content), {
+      uniqueKeys: false,
+    });
+    for (const doc of docs) {
+      if (is.object(doc)) {
+        const includes = getAllIncludeProjects(doc);
+        for (const includeObj of includes) {
+          const dep = extractDepFromIncludeFile(includeObj);
+          if (platform === 'gitlab' && endpoint) {
+            dep.registryUrls = [endpoint.replace(regEx(/\/api\/v4\/?/), '')];
```
Comment (viceice) [inline lib/modules/manager/gitlabci-include/extract.ts:84]: please extract this to outer loop, so we don't need to do the replace again and again

**Review by viceice — APPROVED** [review]

**Pull request merged**